src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -158,21 +232,21 @@ private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax
         private class DeconstructionVariable
         {
             public readonly BoundExpression Single;
-            public readonly ImmutableArray<DeconstructionVariable> Nested;
+            public readonly ArrayBuilder<DeconstructionVariable> Nested;
 
             public DeconstructionVariable(BoundExpression variable)
             {
                 Single = variable;
-                Nested = default(ImmutableArray<DeconstructionVariable>);
+                Nested = null;
             }
 
-            public DeconstructionVariable(ImmutableArray<DeconstructionVariable> variables)
+            public DeconstructionVariable(ArrayBuilder<DeconstructionVariable> variables)
             {
                 Single = null;
                 Nested = variables;
             }
 
-            public bool IsNested => !Nested.IsDefault;
+            public bool IsNested => Nested != null;
```

Comment:
You're right, the name of this property could be clearer. How about `HasNestedVariables`?
`HasNestedDeclarations` is not so good because this structure is also used for d-assignment (which doesn't declare variables, just references them).